crates/solver/src/settlement_submission.rs

```diff
@@ -425,69 +461,110 @@ impl SubmissionError {
                     "transaction cancelled after revert or timeout, hash: {:?}",
                     hash
                 )
             }
-            SubmissionError::SimulationRevert(None) => anyhow!("transaction simulation reverted"),
-            SubmissionError::Disabled(reason) => {
+            Self::SimulationRevert(None) => anyhow!("transaction simulation reverted"),
+            Self::Disabled(reason) => {
                 anyhow!("transaction disabled, reason: {:?}", reason)
             }
-            SubmissionError::Other(err) => err,
+            Self::Other(err) => err,
         }
     }
 
     pub fn is_transaction_mined(&self) -> bool {
         match self {
-            SubmissionError::SimulationRevert(_) => false,
-            SubmissionError::Revert(_) => true,
-            SubmissionError::Timeout => false,
-            SubmissionError::Canceled(_) => true,
-            SubmissionError::Other(_) => false,
-            SubmissionError::Disabled(_) => false,
+            Self::SimulationRevert(_) => false,
+            Self::Revert(_) => true,
+            Self::Timeout => false,
+            Self::Canceled(_) => true,
+            Self::Other(_) => false,
+            Self::Disabled(_) => false,
         }
     }
 }
 
-impl From<web3::Error> for SubmissionError {
+impl From<web3::Error> for SubmissionErrorClass {
     fn from(err: web3::Error) -> Self {
         Self::Other(err.into())
     }
 }
 
-impl From<anyhow::Error> for SubmissionError {
+impl From<anyhow::Error> for SubmissionErrorClass {
     fn from(err: anyhow::Error) -> Self {
         Self::Other(err)
     }
 }
 
-impl From<MethodError> for SubmissionError {
+impl From<MethodError> for SubmissionErrorClass {
     fn from(err: MethodError) -> Self {
         match err.inner {
-            ExecutionError::ConfirmTimeout(_) => SubmissionError::Timeout,
+            ExecutionError::ConfirmTimeout(_) => SubmissionErrorClass::Timeout,
             ExecutionError::Failure(_) | ExecutionError::InvalidOpcode => {
-                SubmissionError::SimulationRevert(None)
+                SubmissionErrorClass::SimulationRevert(None)
             }
-            ExecutionError::Revert(message) => SubmissionError::SimulationRevert(message),
-            _ => SubmissionError::Other(
+            ExecutionError::Revert(message) => SubmissionErrorClass::SimulationRevert(message),
+            _ => SubmissionErrorClass::Other(
                 anyhow::Error::from(err).context("settlement transaction failed"),
             ),
         }
     }
 }
 
+/// An error during settlement submission.
+#[derive(Debug)]
+pub struct SubmissionError {
+    pub inner: SubmissionErrorClass,
+    // Needed for metric purposes
+    pub strategy: &'static str,
+}
+
+impl std::fmt::Display for SubmissionError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{self:?}")
+    }
+}
+
+impl std::error::Error for SubmissionError {}
```

Comment:
The `std::error::Error` implementation doesn't seem to be required for compilation.

Reply:
Irrelevant after latest changes.